cmake/onnxruntime_webassembly.cmake

```diff
@@ -287,7 +287,11 @@ else()
       "SHELL:-s EXPORT_NAME=ortWasm"
     )
     if (onnxruntime_ENABLE_WEBASSEMBLY_SIMD)
-      set_target_properties(onnxruntime_webassembly PROPERTIES OUTPUT_NAME "ort-wasm-simd")
+      if (onnxruntime_ENABLE_TRAINING_APIS)
```

Comment:
no, we don't modify the target names. instead we rename them in the copy steps. see how we do for JSEP.

Reply:
Ah, why is having the user / a script rename the WASM artifacts in the copy steps preferred?

Reply:
it simplifies the cmake file. the more variants we have, the more if-then-else we need to write in this file. it's easy to miss a certain combination ( for example, in this commit you forgot "ort-wasm-simd-threaded-training" as you planned to write )

Reply:
Ah I didn't add that variant because I had thought that web training wasn't going to support multi-threading -- but I see what you mean about adding complexity to the cmake file with nested if-else statements. 

I refactored the target naming in the cmake file in the last commit because I slightly prefer that build steps don't have the added complexity of renaming during the copy steps and also because more WASM artifacts will probably be required over time.